posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,322 @@
+
+(The file 'org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar' is a runnable jar file that specifies 'org.eclipse.transformer.jakarta.JakartaTransformer' as the main class and uses the necessary jars from the 'libs' folder in its class path.)
+
+The transformer can be run using custom rules by specifying additional options on the command line.  The additional options specify local rules files that the transformer uses instead of the default rules files.
+
+For example, to use Open Liberty rules data, which can be more up-to-date than the default rules data, download link:https://github.com/OpenLiberty/open-liberty/tree/integration/dev/wlp-jakartaee-transform/rules[Open Liberty Rules].  Then pull in the downloaded rules data by adding options "-tr", "-td", and "-tf", to the command line invocation:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location> \
+    -tr ${rules}/jakarta-renames.properties \
+    -td ${rules}/jakarta-direct.properties \
+    -tf ${rules}/jakarta-xml-master.properties
+--
+
+The "-tr" option specifies a properties file which contains package rename data.  The "-td" option specifies a properties file which contains java direct string replacement data.  The "-tf" option specifies a master properties file which links to other properties files.  Each of the linked properties files contains replacement data for text-like files matching specific patterns, as specified in the master properties file.
+
+These three command line options are most important for application and test developers.  Other command line options are available, but are less important.  For example, the option "-tb" is used to specify data for transforming bundle archives, while the option "-ts" is used to specify which target files are to be transformed.  For more information about command line options, run the transformer with either of the options "-usage" or "-help":
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    -usage
+--
+
+=== Transforming server configurations
+
+Transformation of a server configuration is only necessary when "javax" prefixed package names are exposed in the server configuration.  One case is JMS activation specification elements.  For example:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="javax.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+This must be transformed to:
```

Comment:
Avoid using "This" as the subject:

This configuration must be transformed to:

Reply:
also, per my previous comment, should make clear whether this is something the test framework does, or something the user has to do

Reply:
Several updates were made to this section.